django/librarian/models.py

```diff
@@ -411,6 +411,11 @@ def delete(self, *args, **kwargs):
             self.remove_from_vector_store()
         except Exception as e:
             logger.error(f"Failed to remove document from vector store: {e}")
+        file = self.file
+        if file:
+            self.file = None
+            self.save()
+            file.safe_delete()
```

Comment:
This could probably also be accomplished via the post_delete handler but this seems to be working. I could / should add tests.